lib/cover.nix

```diff
@@ -0,0 +1,98 @@
+{ stdenv, lib, haskellLib, pkgs }:
+
+{ name, version, library, tests, plan }:
+
+let
+  buildWithCoverage = builtins.map (d: d.covered);
+  runCheck = builtins.map (d: haskellLib.check d);
+
+  libraryCovered    = library.covered;
```

Comment:
This isn't quite strong enough: you might also need to build any internal libraries with coverage. Generally, I think this an argument for requiring the user to specify the coverage options, then it's on them to get this right!